Firebase/Core/FIRSwizzlingCaches.mm

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "FIRSwizzlingCaches.h"
+
+
+// Use C++ collections because Class and SEL are not ObjC types. We also want to maximize speed.
```

Comment:
Are C++ collections measurably faster than Objective C collections?

It would be nice to keep Objective C++ out of Core

Reply:
In this case, yes they are. At SDK start, we could swizzle potentially hundreds of methods. Based on https://gist.github.com/rsms/703730 and internal testing from a long time ago, we're seeing about a 20% performance difference between the collection types. Since this will be put into its own pod, Core won't ever get ObjC++ into master (probably).